ragstack-e2e-tests/e2e_tests/langchain/rag_application.py

```diff
@@ -213,10 +230,38 @@ def run_conversational_rag(
         memory=memory,
         callbacks=[ConsoleCallbackHandler()],
     )
-    result = conversation({"question": "what is MyFakeProductForTesting?"})
-    logging.info("First result: " + str(result))
 
-    result = conversation({"question": "and when was it released?"})
-    logging.info("Second result: " + str(result))
+    tracer = LangChainTracer(project_name="default")
+
+    with callbacks.collect_runs() as cb:
+        result = conversation.invoke(
+            {"question": "what is MyFakeProductForTesting?"},
+            config={"callbacks": [tracer]},
+        )
+        run_id = cb.traced_runs[0].id
+        record_langsmith_sharelink(LANGSMITH_CLIENT, run_id, record_property)
+        logging.info("First result: " + str(result))
+
+    with callbacks.collect_runs() as cb:
```

Comment:
Split these up because it's easier. You could probably do something like `len` of runs after first invocation, then get `traced_runs[len]` for second invocation, but this worked so left it as is.